src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -179,49 +260,121 @@ private static CollectionSettings createCollectionSettings(
+        return new CollectionSettings(
+            collectionName, new VectorConfig(true, vectorSize, function, null), null);
+      } else {
+        return new CollectionSettings(collectionName, VectorConfig.notEnabledVectorConfig(), null);
+      }
     } else {
-      String vectorizeServiceName = null;
-      String modelName = null;
-      JsonNode commentConfig;
+      JsonNode commentConfigNode;
       try {
-        commentConfig = objectMapper.readTree(comment);
+        commentConfigNode = objectMapper.readTree(comment);
       } catch (JsonProcessingException e) {
         // This should never happen, already check if vectorize is a valid JSON
         throw new RuntimeException("Invalid json string, please check 'options' configuration.", e);
       }
-      JsonNode vectorizeConfig = commentConfig.path("vectorize");
-      if (!vectorizeConfig.isMissingNode()) {
-        vectorizeServiceName = vectorizeConfig.path("service").textValue();
-        JsonNode optionsNode = vectorizeConfig.path("options");
-        modelName = optionsNode.path("modelName").textValue();
-        if (!(vectorizeServiceName != null
-            && !vectorizeServiceName.isEmpty()
-            && modelName != null
-            && !modelName.isEmpty())) {
-          // This should never happen, VectorizeConfig check null, unless it fails
-          throw new JsonApiException(
-              VECTORIZECONFIG_CHECK_FAIL,
-              "%s, please check 'vectorize' configuration."
-                  .formatted(VECTORIZECONFIG_CHECK_FAIL.getMessage()));
+
+      // new table comment design, with collection as top-level key
+      if (commentConfigNode.has(TableCommentConstants.TOP_LEVEL_KEY)) {
+        JsonNode collectionNode = commentConfigNode.get(TableCommentConstants.TOP_LEVEL_KEY);
+        JsonNode collectionOptionsNode = collectionNode.get(TableCommentConstants.OPTIONS_KEY);
+        VectorConfig vectorConfig = VectorConfig.notEnabledVectorConfig();
+        JsonNode vector = collectionOptionsNode.path(TableCommentConstants.COLLECTION_VECTOR_KEY);
+        if (!vector.isMissingNode()) {
+          vectorConfig = VectorConfig.fromJson(vector);
+        }
+        IndexingConfig indexingConfig = null;
+        JsonNode indexing =
+            collectionOptionsNode.path(TableCommentConstants.COLLECTION_INDEXING_KEY);
+        if (!indexing.isMissingNode()) {
+          indexingConfig = IndexingConfig.fromJson(indexing);
+        }
+        return new CollectionSettings(collectionName, vectorConfig, indexingConfig);
+      } else {
+        VectorConfig vectorConfig = new VectorConfig(vectorEnabled, vectorSize, function, null);
+        // backward compatibility for old indexing table comment
+        IndexingConfig indexingConfig = null;
+        JsonNode indexing = commentConfigNode.path(TableCommentConstants.COLLECTION_INDEXING_KEY);
+        if (!indexing.isMissingNode()) {
+          indexingConfig = IndexingConfig.fromJson(indexing);
```

Comment:
Could you add more explanations to this block of code? I know we only stored "indexConfig" at the very beginning and now we have to store info in "comment". As time goes by, we may keep changing the "comment" and it may be confusing and hard to maintain.

Reply:
Sure, will add more comments on it.
Currently, there is no other way of doing this. For indexing backwards compatibility, extra logic can not be avoid.